sourced/ml/repo2/wmhash.py

```diff
@@ -144,6 +145,36 @@ def inspect(self, uast):
             yield key
 
 
+class StructureBagExtractor(BagsExtractor):
```

Comment:
Really like this. Let's kill `extract` and `inspect` and merge `StructureBagExtractor` into `BagsExtractor`. Plus replace `self.uast2bag.uast_to_bag` with

```
def uast_to_bag(self, uast):
    raise NotImplementedError()
```